### PR TITLE
vaccine dose taken by day

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
[Number of people received 1st dose of COVID-19 vaccine by day #2559](https://github.com/beyondessential/tupaia-backlog/issues/2559)
[Number of people received 2nd dose of COVID-19 vaccine by day#2560
](https://github.com/beyondessential/tupaia-backlog/issues/2560)
### Changes:

### CODE REVIEW

File: packages/database/src/migrations/20210325030144-VaccineDoseTakenByDayVaccineTracking-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      name: `Number of people received ${doseName} dose of COVID-19 vaccine by day`,
+      type: 'chart',
+      chartType: 'bar',
+      periodGranularity: 'day',
+    },
+  };
+};
+
+exports.up = async function (db) {
+  for (const { code: dashboardGroupCode, hierarchyLevel } of dashboardGroups) {
```
Comment (chris-bes) [inline packages/database/src/migrations/20210325030144-VaccineDoseTakenByDayVaccineTracking-modifies-data.js:87]: Hmmm would this migration work? It doesn't seem to have `hierarchyLevel` in a dashboard group object?

File: packages/database/src/migrations/20210325030144-VaccineDoseTakenByDayVaccineTracking-modifies-data.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    for (const { code: dataElementCode, name: doseName } of dataElements) {
+      const report = generateReport(dashboardGroupCode, hierarchyLevel, doseName, dataElementCode);
```
Comment (chris-bes) [inline packages/database/src/migrations/20210325030144-VaccineDoseTakenByDayVaccineTracking-modifies-data.js:89]: `generateReport()` also receives `aggregationEntityType`, but it seems like it's never passed in from here or anywhere else?

**Review by chris-bes — CHANGES_REQUESTED** [review]

**Comment by chris-bes** [comment]
@billli0 I fixed up the migration so maybe you can re-review it?

**Comment by billli0** [comment]
LGTM

**Review by chris-bes — APPROVED** [review]

**Pull request merged**